### PR TITLE
Bugfix/356 messagebrokerpublisher

### PR DESCRIPTION
**Pull request by Maxvanhattum**
## Description

Messagebrokerpublisher is a class library, but it retrieved connection values by it self. These should be passed in by the user of the library. 
Now when used in the API, the connection values are in appsettings.development.json, and then passed into the connection factory by dependecy injection.
Other contexts should now provide the RabbitMQConnectionFactory with the connection values themselves.

## Type of change
Might be breaking if new features outside of the API AND not yet in development utilized this class, instantiating a RabbitMQConnectionFactory by themselves, they should now pass in hostname, username and password.
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [ ] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Publish task to the messagebroker or connect to the messagebroker, see that the connection is now established based on the values in appsettings.development OR if present from environment variables (overwriting the appsettings.).  

1. Go to MessageBrokerPublisher class
2. Use Notificationsender class.
OR

1. Use notificationsystem to subscribe to subject in Messagebroker

## Link to issue

Closes: #356

### CODE REVIEW

File: API/Configuration/Config.cs [hunk 1]
```diff
@@ -143,4 +151,37 @@ public class SwaggerConfig
...
+        public string Username { get; set; }
+
+        /// <summary>
+        /// Gets or sets the username.
+        /// </summary>
+        /// <value>
+        /// The username.
+        /// </value>
+        [Required]
+        public string Password { get; set; }
```
Comment (matir8) [inline API/Configuration/Config.cs:185]: Change `username` to `password` here in the comments

File: API/HelperClasses/EmailSender.cs [hunk 2]
```diff
@@ -23,16 +23,14 @@ public interface IEmailSender
     /// </summary>
     public class EmailSender : IEmailSender
```
Comment (matir8) [inline API/HelperClasses/EmailSender.cs:24]: Now that `API/HelperClasses/EmailSender.cs` has changed, maybe it is a good idea to add some small guide on how to use it correctly

**Review by matir8 — CHANGES_REQUESTED** [review]
Looks good. I have a couple of minor comments

**Review by MeesvanStraten — DISMISSED** [review]
Looks nice the way the user of the messagebroker now needs to supply the credentials from appconfig.json. Other then the little things @matir8  mentioned all looks good!

**Review by MeesvanStraten — APPROVED** [review]

**Review by matir8 — APPROVED** [review]
Thanks for addressing my feedback! This can be merged now. :rocket:

**Pull request merged**